File: wwwroot/js/tripsController.js
(function() {
    "use strict";

    angular.module("app-trips")
        .controller("tripsController", tripsController);

    function tripsController($http) {
        var vm = this;

        vm.trips = [];
        vm.newTrip = {};
        vm.errorMessage = "";
        vm.isBusy = true;

        vm.addTrip = function() {
            vm.isBusy = true;
            vm.errorMessage = "";

            $http.post("/api/trips", vm.newTrip)
                .then(function(response) {
                    // Success
                    vm.trips.push(response.data);
                    vm.newTrip = {};
                }, function() {
                    // Failure
                    vm.errorMessage = "Failed to save new trip";
                })
                .finally(function() {
                    vm.isBusy = false;
                });
        };

        vm.deleteTrip = function(tripId) {
            vm.isBusy = true;
            vm.errorMessage = "";

            $http.delete("/api/trips/" + tripId)
                .then(function() {
                    // Success
                    getTrips();
                }, function() {
                    // Failure
                    vm.errorMessage = "Failed to delete trip";
                })
                .finally(function() {
                    vm.isBusy = false;
                });
        };

        var getTrips = function() {
            $http.get("/api/trips")
                .then(function(response) {
                    // Success
                    // vm.trips = response.data;
                    angular.copy(response.data, vm.trips);
                }, function() {
                    // Failure
                    vm.errorMessage = "Failed to load data";
                })
                .finally(function() {
                    vm.isBusy = false;
                });
        }

        getTrips();
    }
})();

File: Controllers/Api/TripsController.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using AutoMapper;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Logging;
using TheWorld.Entities;
using TheWorld.ViewModels;

namespace TheWorld.Controllers.Api
{
    [Route("api/trips")]
    [Authorize]
    public class TripsController : Controller
    {
        private IWorldRepository _repository;
        private ILogger<TripsController> _logger;

        public TripsController(IWorldRepository repository, ILogger<TripsController> logger)
        {
            _repository = repository;
            _logger = logger;
        }

        //[HttpGet("api/trips")]
        [HttpGet("")]
        public IActionResult Get()
        {
            try
            {
                var results = _repository.GetTripsByUsername(User.Identity.Name);
                return Ok(Mapper.Map<IEnumerable<TripViewModel>>(results));
            }
            catch (Exception ex)
            {
                _logger.LogError($"Failed to get All Trips: {ex}");
                return BadRequest("Error occurred");
            }
        }

        //[HttpPost("api/trips")]
        [HttpPost("")]
        public async Task<IActionResult> Post([FromBody]TripViewModel theTrip)
        {
            if (ModelState.IsValid)
            {
                var newTrip = Mapper.Map<Trip>(theTrip);
                newTrip.UserName = User.Identity.Name;
                _repository.AddTrip(newTrip);

                if (await _repository.SaveChangesAsync())
                {
                    return Created($"api/trips/{theTrip.Name}", Mapper.Map<TripViewModel>(newTrip));
                }
            }

            // return BadRequest("Bad data");
            //return BadRequest(ModelState);      // not recommended for public api
            return BadRequest("Failed to save the trip");
        }

        [HttpDelete("{tripId}")]
        public async Task<IActionResult> Delete(int tripId)
        {
            if (ModelState.IsValid)
            {
                _repository.RemoveTripById(tripId);

                if (await _repository.SaveChangesAsync())
                {
                    return Ok();
                }
            }

            return BadRequest("Failed to save the trip");
        }
    }
}